dask/array/routines.py

```diff
@@ -1870,8 +1870,11 @@ def roll(array, shift, axis=None):
         raise ValueError("Must have the same number of shifts as axes.")
 
     for i, s in zip(axis, shift):
-        s = -s
-        s %= result.shape[i]
+        if result.shape[i] == 0:
+            s = 0
+        else:
+            s = -s
+            s %= result.shape[i]
```

Comment:
Small nit - I'd find this easier to read (and I think it should be equivalent). Feel free to ignore.

```suggestion
        shape = result.shape[i]
        s = 0 if shape == 0 else -s % shape
```

Reply:
I like it!